packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -112,8 +112,17 @@ export default function MainFeedLayout({
   const { flags } = useContext(FeaturesContext);
   const feedVersion =
     parseInt(getFeatureValue(Features.FeedVersion, flags), 10) || 1;
-  const [defaultFeed] = usePersistentState('defaultFeed', null, 'popular');
+  const [defaultFeed, setDefaultFeed] = usePersistentState(
+    'defaultFeed',
+    null,
+    'popular',
+  );
   const feedName = feedNameProp === 'default' ? defaultFeed : feedNameProp;
+
+  if (defaultFeed !== null && feedName !== null && feedName !== defaultFeed) {
+    setDefaultFeed(feedName);
+  }
```

Comment:
Not sure I understand the condition here and also make sure to use `useEffect` for this purpose